derive/src/to_client.rs

```diff
@@ -0,0 +1,280 @@
+use crate::rpc_attr::AttributeKind;
+use crate::to_delegate::{generate_where_clause_serialization_predicates, MethodRegistration};
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use syn::punctuated::Punctuated;
+use syn::Result;
+
+pub fn generate_client_module(methods: &[MethodRegistration], item_trait: &syn::ItemTrait) -> Result<TokenStream> {
+	let client_methods = generate_client_methods(methods)?;
+	let generics = &item_trait.generics;
+	let where_clause = generate_where_clause_serialization_predicates(&item_trait, true);
+	let where_clause2 = where_clause.clone();
+	let markers = generics
+		.params
+		.iter()
+		.filter_map(|param| match param {
+			syn::GenericParam::Type(syn::TypeParam { ident, .. }) => Some(ident),
+			_ => None,
+		})
+		.enumerate()
+		.map(|(i, ty)| {
+			let field_name = "_".to_string() + &i.to_string();
+			let field = Ident::new(&field_name, ty.span());
+			(field, ty)
+		});
+	let (markers_decl, markers_impl): (Vec<_>, Vec<_>) = markers
+		.map(|(field, ty)| {
+			(
+				quote! {
+					#field: std::marker::PhantomData<#ty>
+				},
+				quote! {
+					#field: std::marker::PhantomData
+				},
+			)
+		})
+		.unzip();
+	Ok(quote! {
+		/// The generated client module.
+		pub mod gen_client {
+			use super::*;
+			use _jsonrpc_core::{
+				Call, Error, ErrorCode, Id, MethodCall, Params, Request,
+				Response, Version,
+			};
+			use _jsonrpc_core::futures::{future, Future, Sink};
+			use _jsonrpc_core::futures::sync::oneshot;
+			use _jsonrpc_core::serde_json::{self, Value};
+			use jsonrpc_client::{RpcChannel, RpcError, RpcFuture, RpcMessage};
```

Comment:
Could we use `proc_macro_crate::crate_name("jsonrpc-client")` to be consistent with other imports and protect against crate renames